src/ConstructiveSolidGeometry/VolumePrimitives/Cone.jl

```diff
@@ -1,53 +1,79 @@
 struct Cone{T,TR,TP,TZ} <: AbstractVolumePrimitive{T}
-    rtop::TR
-    rbot::TR
+    r::TR #if not a Tuple, then Cone is a Tube
     φ::TP
     z::TZ
-    
+    #if r is a Tuple, the first entry refers to the r-interval at the bottom, the second one to the r-interval at the top
     function Cone( ::Type{T},
-                   rtop::Union{T, <:AbstractInterval{T}},
-                   rbot::Union{T, <:AbstractInterval{T}},
+                   r::Union{T, <:AbstractInterval{T}, Tuple{T,T}, Tuple{<:AbstractInterval{T},<:AbstractInterval{T}}},
```

Comment:
Is `Tuple{T, <:AbstractInterval{T}}` not also an option?
But make things probably just to complicated?

Reply:
I excluded this option because if one `Interval` starts at `0` and the other one doesn't, we still have to determine the `r`-Interval for a given `z` anyhow. Then it is easier to stick to only deal with intervals.

Reply:
Alright, than just leave it like that.